### PR TITLE
Override Message default equalty operator

### PR DESCRIPTION
**Pull request by busslina**
### (Change 1): Compares type, chatId and messageId.

### CODE REVIEW

**Comment by busslina** [comment]
### Resolved null ErrorResponse parameters field

Example of error stack trace:

http response json data: {ok: false, error_code: 400, description: Bad Request: message can't be deleted}

Unhandled exception:
type 'Null' is not a subtype of type 'Map<String, dynamic>' in type cast
#0      _$ErrorResponseFromJson (package:teledart/src/telegram/model.g.dart:3578:30)
https://github.com/DinoLeung/TeleDart/pull/1      new ErrorResponse.fromJson (package:teledart/src/telegram/models/response.dart:78:7)
https://github.com/DinoLeung/TeleDart/pull/2      HttpClient._parseResponse (package:teledart/src/util/http_client.dart:43:51)
https://github.com/DinoLeung/TeleDart/pull/3      _rootRunUnary (dart:async/zone.dart:1399:47)
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/pull/4      Telegram.deleteMessage (package:teledart/src/telegram/telegram.dart:2346:12)
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/pull/5      TelegramMessageRegister.removeMessages.<anonymous closure> (package:busslina_dart_rest_lib/core/module>
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/pull/6      BasicLock.synchronized (package:synchronized/src/basic_lock.dart:33:16)
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/pull/7      _TelegramHandler._getMenuCommandHandler (package:busslina_dart_rest_lib/core/modules/system.module.dar>
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/pull/8      TelegramBot.onCommand.<anonymous closure> (package:busslina_dart_telegram_lib/src/core/app/telegram_bo>
<asynchronous suspension>
https://github.com/DinoLeung/TeleDart/issues/9      BasicLock.synchronized (package:synchronized/src/basic_lock.dart:33:16)
<asynchronous suspension>
autodia-server.service: Main process exited, code=exited, status=255/EXCEPTION

File: lib/src/teledart/fetch/long_polling.dart [hunk 1]
```diff
@@ -108,9 +108,17 @@ class LongPolling extends AbstractUpdateFetcher {
     return Future.value();
   }
 
-  void _onRecursivePollingHttpError(HttpClientException error) {
+  Future<void> _onRecursivePollingHttpError(HttpClientException error) async {
     if (error.isTooManyRequests()) {
       retryDelay = error.parameters?.retryAfter_ ?? retryDelay;
+      _onRecursivePollingError(error);
+    } else if (error.code == 409) {
```
Comment (DinoLeung) [inline lib/src/teledart/fetch/long_polling.dart:115]: [409 error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) is a conflict error, and it does not seem to be a retry-able error.

File: lib/src/telegram/models/audio.dart [hunk 2]
```diff
@@ -52,4 +52,18 @@ class Audio {
 
   factory Audio.fromJson(Map<String, dynamic> json) => _$AudioFromJson(json);
   Map<String, dynamic> toJson() => _$AudioToJson(this);
+
+  @override
+  String toString() => '''Audio
```
Comment (DinoLeung) [inline lib/src/telegram/models/audio.dart:57]: what are these `toString` methods for?

File: lib/src/telegram/models/message.dart [hunk 3]
```diff
@@ -212,4 +212,13 @@ class Message {
       return null;
     }
   }
+
+  @override
```
Comment (DinoLeung) [inline lib/src/telegram/models/message.dart:216]: this is cool

File: lib/src/telegram/models/response.dart [hunk 4]
```diff
@@ -65,7 +65,7 @@ class ErrorResponse {
   bool ok;
   String description;
   int errorCode;
-  ResponseParameters parameters;
+  ResponseParameters? parameters;
```
Comment (DinoLeung) [inline lib/src/telegram/models/response.dart:68]: good catch

**Review by DinoLeung — COMMENTED** [review]
@busslina thanks for submitting this PR, it seems that it is addressing multiple thing here. Will be closing this PR, and create a new one to address the issue which is the `ResponseParameters` type error.

File: lib/src/teledart/fetch/long_polling.dart (hunk 1 above)
Comment (busslina) [reply lib/src/teledart/fetch/long_polling.dart:115]: Okay, but like in 429 error, it must be a way to catch it and not let the whole app crash

File: lib/src/telegram/models/audio.dart (hunk 2 above)
Comment (busslina) [reply lib/src/telegram/models/audio.dart:57]: These are methods to debug that I use internally on my fork

File: lib/src/teledart/fetch/long_polling.dart (hunk 1 above)
Comment (DinoLeung) [reply lib/src/teledart/fetch/long_polling.dart:115]: this [PR](https://github.com/DinoLeung/TeleDart/pull/243/files) has export the http client along with the exception class 👍

**Pull request closed without merging**